lib/seahorn/HornCex.cc

```diff
@@ -220,11 +227,25 @@ namespace seahorn
               << ". Expected .xml, .ll, or .bc.\n";
     }
 
-    if (!SliceOutputFile.empty()) {
-      sliceErrorTrace(F, trace);
-      dumpLLVMBitcode(M, SliceOutputFile.c_str());
+    if (!BmcSliceOutputFile.empty()) {
+      llvm::DenseSet<const llvm::BasicBlock*> region(trace.getRegion());
+      reduceToRegion(F, region, false);
+      dumpLLVMBitcode(M, BmcSliceOutputFile.c_str());
       return true;
     }
+
+    if (!CpSliceOutputFile.empty())
+    {
+      DenseSet<const BasicBlock*> region;
+      for (const CutPoint* cp:cpTrace)
+      for (CutPoint::const_iterator it=cp->succ_begin();it!=cp->succ_end();it++)
+      for (CpEdge::iterator ite=(*it)->begin();ite!=(*it)->end();ite++)
+        region.insert(&*ite);
+      reduceToRegion(F,region,false);
```

Comment:
what is the reason for not deleting the basic blocks that are outside of the trace? Deleting them will enable additional optimizations. Moreover, the blocks are unreachable after the reduction and will be removed by dead code elimination anyways.